File: training-data/r/binary_search.R
binary_search <- function(arr, target) {
  low <- 1
  high <- length(arr)
  while (low <= high) {
    mid <- (low + high) %/% 2
    if (arr[mid] == target) {
      return(mid)
    } else if (arr[mid] < target) {
      low <- mid + 1
    } else {
      high <- mid - 1
    }
  }
  -1
}

sorted <- c(1, 3, 5, 7, 9, 11, 13)
print(binary_search(sorted, 7))
print(binary_search(sorted, 4))

File: training-data/lua/binary_search.lua
local function binarySearch(arr, target)
  local low, high = 1, #arr
  while low <= high do
    local mid = (low + high) // 2
    if arr[mid] == target then
      return mid
    elseif arr[mid] < target then
      low = mid + 1
    else
      high = mid - 1
    end
  end
  return -1
end

local sorted = { 1, 3, 5, 7, 9, 11, 13 }
print(binarySearch(sorted, 7))
print(binarySearch(sorted, 4))
